### PR TITLE
📖Update to use FC and not SFC

### PR DESCRIPTION
**Pull request by Pau1fitz**
DefinitelyTyped/DefinitelyTyped#30364

### CODE REVIEW

**Comment by piotrwitek** [comment]
Hey, that improvement looks cool!

Please remember about: https://github.com/piotrwitek/react-redux-typescript-guide/blob/master/CONTRIBUTING.md

I can lend a hand at weekend to with rebuilding code generation for example etc.

**Comment by piotrwitek** [comment]
Hey @Pau1fitz did it already landed on @types? How about SFC, will it still be supported or is it breaking all SFC usages?

**Comment by Pau1fitz** [comment]
Hey @piotrwitek, I tried to run the changes through the contributing guideline stages over the weekend, but had some issues. they are just deprecated at the moment, but still work. Would be good to update this guide though as a lot of people use this resource. You can see that [this guide](https://github.com/sw-yx/react-typescript-cheatsheet#function-components) uses `FunctionComponent`.

**Comment by piotrwitek** [comment]
Got it, the reason I'm asking is I was trying it like a few days ago and `FC` was not part of react types.

I'll try again

**Comment by Pau1fitz** [comment]
cool, let me know.

**Comment by piotrwitek** [comment]
OK, checked and working fine.

One more thing, could you please change all instances of "Stateful components" to "Class components"? As of now this distinction will make more sense.

File: README.md [hunk 1]
```diff
@@ -172,20 +172,20 @@ const handleChange = (ev: React.MouseEvent<HTMLDivElement>) => { ... }
 
 # Component Typing Patterns
 
-## Stateless Components - SFC
+## Function Components - FC
 
-#### - stateless counter
+#### - functional counter
```
Comment (piotrwitek) [inline README.md]: functional counter is a bit confusing name, let's name it "counter component"

Comment (Pau1fitz) [reply README.md]: sure, sounds good. I just changed it as stateless is no longer the term being used. are you happy for me to do it in this PR, as I said I was having issues with the instructions in the contributing guidelines - it wasn't working  😯

Comment (piotrwitek) [reply README.md]: Yea sure.
Could you also post what was the issue with contributing guide as I would like to fix that.
Thanks

**Comment by Pau1fitz** [comment]
issue with the contributing guide was step 2:

```
# run type-checking in `/playground` folder
npm run tsc
```

Lots of errors:

screenshot 2018-12-12 at 15 56 46

**Comment by piotrwitek** [comment]
> Changes updated.
> 
> The issue with the contributing guide that I am having is in relation to step 2:
> 
> ```
> # run type-checking in `/playground` folder
> npm run tsc
> ```
> Lots of errors:
...

Thanks.
I guess errors are something to do with how you install dependencies, because on clean install there is no errors.

Did you you npm install with `npm` at least v@5.X.X to respect package-lock?

**Comment by Pau1fitz** [comment]
I am using npm v 6.4.1

**Comment by piotrwitek** [comment]
Have you tried to `rm -rf node_modules` and then do a clean install?

**Comment by piotrwitek** [comment]
Rebased your work on master, updated readme-source, updated source files in playground, updated styleguidist webpage.

Merged in #124

**Comment by Pau1fitz** [comment]
thanks @piotrwitek good work

**Comment by piotrwitek** [comment]
@Pau1fitz Thanks for the initial work! I wanted to retain your contribution but when I squashed and merged the PR github didn't honored it. Will need to check some guide how to do it properly next time.

**Comment by Pau1fitz** [comment]
will have to make another PR 😁🤓

**Pull request closed without merging**